pkg/apis/gadget/v1alpha1/types.go

```diff
@@ -21,6 +21,35 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// RunMode defines running mode for the Trace
+// +kubebuilder:validation:Enum=Auto;Manual
+type RunMode string
+
+const (
+	// RunModeAuto automatically starts the trace as soon
+	// as the resource is created.
+	RunModeAuto RunMode = "Auto"
+	// RunModeManual allows the trace to be controlled by the
+	// "gadget.kinvolk.io/operation" annotation.
+	RunModeManual RunMode = "Manual"
+)
+
+// TraceOutputMode define output mode for the Trace
+// +kubebuilder:validation:Enum=Status;Stream;File;ExternalResource
+type TraceOutputMode string
+
+const (
+	// TraceOutputModeStatus means trace prints information when its status changes
+	TraceOutputModeStatus TraceOutputMode = "Status"
+	// TraceOutputModeStream means trace prints information as events arrive
```

Comment:
```suggestion
	// TraceOutputModeStream indicates to stream events. This stream can be accessed through the Stream() api on the gadget tracer manager
```

@alban any suggestion?